### PR TITLE
461 add contract definition page to the api wrapper

### PR DESCRIPTION
**Pull request by SaadEGI**
Related to #461 Add Contract Definition Page to the API-Wrapper

### CODE REVIEW

File: extensions/wrapper/client/src/test/java/de/sovity/edc/client/ContractDefinitionPageApiServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        var contractDefinition = contractDefinitions.get(0);
+        assertThat(contractDefinition.getContractDefinitionId()).isEqualTo(ContractDefinitionTestUtils.CONTRACT_DEFINITION_ID);
+        assertThat(contractDefinition.getContractPolicyId()).isEqualTo(ContractDefinitionTestUtils.CONTRACT_POLICY_ID);
+        assertThat(contractDefinition.getAccessPolicyId()).isEqualTo(ContractDefinitionTestUtils.ACCESS_POLICY_ID);
+    }
+
+
+
+
+
```
Comment (richardtreier) [inline extensions/wrapper/client/src/test/java/de/sovity/edc/client/ContractDefinitionPageApiServiceTest.java:66]: newlines

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/UiResource.java [hunk 2]
```diff
@@ -73,4 +76,11 @@ public TransferHistoryPage transferHistoryPageEndpoint() {
     public AssetDto getTransferProcessAsset(@PathParam("transferProcessId") String transferProcessId) {
         return transferHistoryPageAssetFetcherService.getAssetForTransferHistoryPage(transferProcessId);
     }
+
+    @GET
+    @Path("pages/contract-definition-page")
+    @Produces(MediaType.APPLICATION_JSON)
+    public ContractDefinitionPage contractDefinitionPageEndpoint() {
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/UiResource.java:83]: Maybe the method name should only be contractDefinitionPage? If the above names are with the Endpoint suffix I think we should remove them as these function names are passed into the generated API client libraries

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractDefinitionEntry.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+public class ContractDefinitionEntry {
+
+    @Schema(description = "Access Policy ID", requiredMode = Schema.RequiredMode.REQUIRED)
+    private String accessPolicyId;
+
+    @Schema(description = "Contract Policy ID", requiredMode = Schema.RequiredMode.REQUIRED)
+    private String contractPolicyId;
+
+    @Schema(description = "Criteria for the contract", requiredMode = Schema.RequiredMode.REQUIRED)
+    private List<Criterion> criteria;
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/ContractDefinitionEntry.java:34]: No Core EDC Types should be leaked. We need to add our own objects and do the mapping.

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contracts/ContractDefinitionApiService.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import org.eclipse.edc.connector.spi.contractdefinition.ContractDefinitionService;
+import org.eclipse.edc.spi.query.QuerySpec;
+import java.util.List;
+
+@RequiredArgsConstructor
+public class ContractDefinitionApiService {
+
+
+    private final ContractDefinitionService contractDefinitionService;
+    public List<ContractDefinitionEntry> getContractDefinitions() {
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contracts/ContractDefinitionApiService.java:29]: newlines. format all changed files once, maybe

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/Criterion.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        this.operandRight = operandRight;
+        this.type = type;
+    }
+
+    public Criterion(String operandLeft, String operator, List<String> operandRightArray, CriterionType type) {
+        this.operandLeft = operandLeft;
+        this.operator = operator;
+        this.operandRightArray = operandRightArray;
+        this.type = type;
+    }
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/model/Criterion.java:40]: For sum types, let's still use Lombok, e.g.

```java
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
class MyObject {
   private MyObjectType type;
   private String optionA;
   private String optionB;

   // Convenience Constructors, e.g. for tests. 
   // Notice how the object remains a POJO, these are just static methods

   public static MyObject withOptionA(String optionA) {
      return new MyObject(MyObjectType.OPTION_A, optionA, null);
   }

   public static MyObject withOptionB(String optionB) {
      return new MyObject(MyObjectType.OPTION_B, null, optionB);
   }
}

```

File: extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contracts/ContractDefinitionApiService.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        var definitions = getAllContractDefinitions();
+        return definitions.stream().map(definition -> {
+            var entry = new ContractDefinitionEntry();
+            entry.setContractDefinitionId(definition.getId());
+            entry.setAccessPolicyId(definition.getAccessPolicyId());
+            entry.setContractPolicyId(definition.getContractPolicyId());
+            entry.setCriteria(definition.getSelectorExpression().getCriteria()); // This depends on how the criteria are stored in your system
+            return entry;
+        }).toList();
```
Comment (richardtreier) [inline extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contracts/ContractDefinitionApiService.java:40]: we should add a sorting, maybe createdDate DESC?

**Review by SaadEGI — COMMENTED** [review]
Added Changes

**Pull request closed without merging**